[BE][export] Fix hop tests with flaky memory leak

yiming0416 opened the pull request:
Summary:
As title. Added `torch._dynamo.reset()` for each test

This should fix several flaky tests in `test_hop.py` such as https://github.com/pytorch/pytorch/issues/139073

Test Plan:
```
PYTORCH_TEST_CUDA_MEM_LEAK_CHECK=1 python test/export/test_hop.py TestHOPCUDA.test_serialize_export_scan_simple_cuda_float32
```

Differential Revision: D68506280

ydwu4 commented on test/export/test_hop.py:
```diff
@@ -85,6 +85,7 @@ def forward(self, *args):
             kwargs = inp.kwargs
             ep = export(model, args, kwargs, strict=True)
             self._compare(model, ep, args, kwargs)
+        torch._dynamo.reset()
```
Probably a better way is to do torch._dynamo.reset() in setUp()?

ydwu4 replied:
teardown() would be more close to what we have now.

ydwu4 replied:
Yeah, the cuda memory check appears to be a context manager: https://github.com/pytorch/pytorch/blob/main/torch/testing/_internal/common_utils.py#L3110-L3121. But we still don't know why we're getting a memory leak.. Need look into it more

ydwu4 reviewed (approved):
memory leak checks seems to happen before tearDown is called. So we have to reset dynamo in test body.

yiming0416 committed:
[BE][export] Fix hop tests with flaky memory leak (#145391)

Summary:

As title. Added `torch._dynamo.reset()` for each test

This should fix several flaky tests in `test_hop.py` such as https://github.com/pytorch/pytorch/issues/139073

Test Plan:
```
PYTORCH_TEST_CUDA_MEM_LEAK_CHECK=1 python test/export/test_hop.py TestHOPCUDA.test_serialize_export_scan_simple_cuda_float32
```

Reviewed By: ydwu4

Differential Revision: D68506280

ezyang commented:
So if we can't put it in tearDown, how do we make sure that when people add new tests they don't make the same mistake?
